### PR TITLE
[cli 4.2.4] Preparing for CLI release

### PR DESCRIPTION
**Pull request by wrwg**
- Updating CHANGELOG
- Marking Move 2.1 stable and the default

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Move Compiler
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by wrwg** [comment]
* **#15027** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/15027?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @wrwg and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by rahxephon89** [comment]
Should we use language version 2.1 when specifying `move-2`: https://github.com/aptos-labs/aptos-core/blob/a822e6518b85881d337967342659d663071dea25/crates/aptos/src/common/types.rs#L1155? @wrwg @brianrmurphy

**Review by 0xmaayan — APPROVED** [review]
LGTM

**Review by rahxephon89 — APPROVED** [review]

File: crates/aptos/src/common/types.rs [hunk 1]
```diff
@@ -1181,7 +1181,7 @@ pub struct MovePackageDir {
     pub language_version: Option<LanguageVersion>,
 
     /// Select bytecode, language version, and compiler to support Move 2:
-    /// Same as `--bytecode_version=7 --language_version=2.0 --compiler_version=2.0`
+    /// Same as `--bytecode_version=7 --language_version=2.1 --compiler_version=2.0`
```
Comment (vineethk) [inline crates/aptos/src/common/types.rs:1184]: In the attributes for the field `language_version` above, we have `default_value_if("move_2", "true", "2.0")`: should we change this to `default_value_if("move_2", "true", "2.1")`?

**Pull request merged**